### DISCUSSION TITLE
How can I parameterize my assets

### DISCUSSION THREAD

**Discussion by yan91083** [dagster-io/dagster#17644] [Q&A] [answered]
Hi guys? 

I have a question about assets and ops. I have 10 files to be processed by 3 ops and the results could be 10*3=30 files. Could I just define 10 assets and 3 ops and combine them with parameters? Or should I define 30 assets?

Thanks!

**Comment by sryza** [comment, marked as answer]
Hey @yan91083 . One way to deal with this would be to use partitions. Here's a discussion that covers this in more detail: https://github.com/dagster-io/dagster/discussions/12061.
